src/pkg/cli/tail.go

```diff
@@ -132,43 +156,26 @@ type CancelError struct {
 	error
 }
 
-func (cerr *CancelError) Error() string {
-	cmd := "tail --since " + cerr.Since.UTC().Format(time.RFC3339Nano)
-	if len(cerr.Services) > 0 {
-		cmd += " --name " + strings.Join(cerr.Services, ",")
-	}
-	if cerr.Etag != "" {
-		cmd += " --etag " + cerr.Etag
-	}
-	if cerr.Verbose {
-		cmd += " --verbose"
-	}
-	return cmd
+func (cerr CancelError) Error() string {
+	return cerr.String()
 }
 
-func (cerr *CancelError) Unwrap() error {
+func (cerr CancelError) Unwrap() error {
 	return cerr.error
 }
 
-func Tail(ctx context.Context, loader client.Loader, provider client.Provider, params TailOptions) error {
-	projectName, err := client.LoadProjectNameWithFallback(ctx, loader, provider)
-	if err != nil {
-		return err
-	}
-	if params.Project == "" {
-		params.Project = projectName
-	}
+func Tail(ctx context.Context, provider client.Provider, params TailOptions) error {
```

Comment:
nitpick, but I don't love hiding `ProjectName` inside `TailOptions`. The other values in that struct are optional, but `ProjectName` is required. I think we should promote it to be part of the `Tail` parameter list.
```suggestion
func Tail(ctx context.Context, provider client.Provider, projectName string, params TailOptions) error {
```